keps/sig-node/4265-proc-mount/README.md

```diff
@@ -0,0 +1,485 @@
+	- "/sys/firmware",
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: `kubelet_pod_start_sli_duration_seconds`
+  - [Optional] Aggregation method:
+  - Components exposing the metric: kubelet
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+- A CRI implementation that supports this feature
+    - All supported versions currently do.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+ProcMountType in the pod spec
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Negligible
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+Potentially a malicious user given access and running with a root container in the host context could mess with the host processes.
+PSS has already been configured to mitigate this by required a user be in a privileged namespace to get access to the field.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No effect
+
+###### What are other known failure modes?
+
+Malicious user, protected by PSS.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+The field can be unset in a pod spec (or feature gate turned off) to see if SLOs met after the feature is disabled for pods.
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+2018-05-07: k/community update opened
+2018-05-27: k/kubernetes PR merged with support.
+2023-10-02: KEP opened and targeted at Beta
+
+## Drawbacks
+
+
+## Alternatives
```

Comment:
LXD uses an approach where it mounts an additional `/dev/.lxc/proc` and `/dev/.lxc/sys`, the advantage being the kernel sees an unmasked proc filesystem (it's fine it's mounted anywhere, not just the root). It then uses AppArmor to deny user access to the paths -- meaning it can offer this feature (`security.nesting=true` in LXD terms) regardless of what paths are masked.

I believe an approach like this has benefits -- it means that /sys can also be mounted in unprivileged containers (because usually `/sys` is masked by `/sys/fs/cgroup` anyway). I begun writing a KEP targeting 1.30 (I haven't published it yet, was waiting for 1.29 KEP to be over, as I wasn't expecting this to get rushed at the last minute).

Reply:
> (it's fine it's mounted anywhere, not just the root).

this is a weird limitation... thanks for the context, I'll add the alternative, and we can chat about alternatives. it's possible the runtimes could do something fancy like that instead of masking...